src/main/java/post_reply_user/Post.java

```diff
@@ -0,0 +1,104 @@
+package post_reply_user;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+public class Post {
+    common_User auther;
+    String content;
+    int likes;
+    String time;
+    ArrayList<common_User> replyedBy;
+    ArrayList<Reply> totalReply;
+    ArrayList<common_User> likedBy;
+
+    public void setAuther(common_User auther) {
+        this.auther = auther;
+    }
+
+    public void setContent(String content) {
+        this.content = content;
+    }
+
+    public void setTime() {
+        this.time = String.valueOf(new Date());
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
+        this.time = dateFormat.format(time);
+    }
+
+    public void setReply_by(){
+        this.replyedBy = new ArrayList<common_User>();
+        this.totalReply = new ArrayList<Reply>();
+
+    }
+
+    public void like_this(common_User user){
+        this.likedBy.add(user);
+        this.likes += 1;
+
+    }
+
+    public ArrayList<common_User> getLike_by() {
+        return likedBy;
+    }
+
+    public common_User getAuther(){
+        return auther;
+    }
+
+    public String getContent() {
+        return content;
+    }
+
+    public String getTime() {
+        return time;
+    }
+
+    public ArrayList<Reply> gettotalReply() {
+        return this.totalReply;
+    }
+
+    public void like_post(common_User user_2){
+        if (this.likedBy.contains(user_2)){
+            this.likedBy.remove(user_2);
+        }
+        else {
+            this.likedBy.add(user_2);
+        }
+    }
+
+    public void add_Reply(Reply reply){
+        this.totalReply.add(reply);
+    }
+
+    public void delete_Reply(Reply reply){
+        this.totalReply.remove(reply);
+    }
+
+    public ArrayList<Reply> top_three(){
+        ArrayList<Reply> result = new ArrayList<Reply>();
+        result.add(this.totalReply.get(0));
+        for (int i = 1; i < this.totalReply.size(); i ++){
+            for (int j = 0; j <= result.size(); j ++){
+                if (this.totalReply.get(i).likes <= result.get(j).likes){
```

Comment:
Mark here.
Assume index 0 reply has the most likes, and we have an ArrayList which contains likes [10,1,5, 7, 8]
Seems like the 10 will first go to the result list [10], and followed by 1, [10, 1], and now no replies will go to the result list because any of 5,7,8 is greater than 1.

Reply:
Agree that this insertion sort has some issues
It looks like it is trying to sort in ascending order instead of descending
And also it throws index out of bounds error on certain inputs

Something like this should probably work (with minimal changes to the original code)
```
    public ArrayList<Reply> sortRepliesByLikes(){
        ArrayList<Reply> result = new ArrayList<Reply>();
        result.add(this.totalReply.get(0));
        for (int i = 1; i < this.totalReply.size(); i ++){
            int j;
            for (j = 0; j <= result.size() - 1; j ++){
                if (this.totalReply.get(i).likes > result.get(j).likes){
                    break;
                }
            }
            result.add(j, this.totalReply.get(i));
        }
        return result;
    }
```

Alternatively, implement a custom comparator and use ArrayList.sort()
https://stackoverflow.com/questions/2784514/sort-arraylist-of-custom-objects-by-property
```
    public ArrayList<Reply> sortRepliesByLikes(){

        // make a deep copy
        ArrayList<Reply> result = new ArrayList<Reply>(totalReply);
        // sort by descending order
        result.sort((o1, o2) -> o2.likes - o1.likes);

        return result;
    }
```